[dav1d] should not depend on `nasm` on arm64

### Is your feature request related to a problem? Please describe.

When building the dav1d port on arm64 (both AArch64 Linux and macOS), the build fails with the following if the `nasm` package is not installed:

```
CMake Error at scripts/cmake/vcpkg_find_acquire_program.cmake:166 (message):
  Could not find nasm.  Please install it via your package manager:

      sudo apt-get install nasm
Call Stack (most recent call first):
  buildtrees/versioning_/versions/dav1d/0349d1196602b2983b1171647b30b5a9379920c6/portfile.cmake:9 (vcpkg_find_acquire_program)
  scripts/ports.cmake:191 (include)
```

This dependency is used only in x86 builds of the port, otherwise it uses GAS or Clang's integrated assembler by invoking the C compiler binary.

### Proposed solution

Similarly to ports like x264, the call to `vcpkg_find_acquire_program(NASM)` should be gated behind a check for x86:

https://github.com/microsoft/vcpkg/blob/7aeffc91033ad35cc4e2c152f213a866ec6c11ac/ports/x264/portfile.cmake#L58-L75

`dav1d` similarly supports gaspp when targeting arm/arm64 Windows.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

--- a/ports/dav1d/portfile.cmake
+++ b/ports/dav1d/portfile.cmake
@@ -6,18 +6,16 @@ vcpkg_from_gitlab(
     SHA512 7f02c1e65fb79352a159cc60346e02d5844c01d1228f860f802c553799ae0213e76ccc81ff427913e89bd3cfcf0185ea063cf3b6f7c4242faa76962ebd661e5e
 )
 
-vcpkg_find_acquire_program(NASM)
-get_filename_component(NASM_EXE_PATH ${NASM} DIRECTORY)
-vcpkg_add_to_path(${NASM_EXE_PATH})
-
-if (VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64")
-    if(VCPKG_TARGET_IS_WINDOWS)
-        vcpkg_find_acquire_program(GASPREPROCESSOR)
-        foreach(GAS_PATH ${GASPREPROCESSOR})
-            get_filename_component(GAS_ITEM_PATH ${GAS_PATH} DIRECTORY)
-            set(ENV{PATH} "$ENV{PATH}${VCPKG_HOST_PATH_SEPARATOR}${GAS_ITEM_PATH}")
-        endforeach(GAS_PATH)
-    endif()
+if (VCPKG_TARGET_ARCHITECTURE STREQUAL "x86" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+    vcpkg_find_acquire_program(NASM)
+    get_filename_component(NASM_EXE_PATH ${NASM} DIRECTORY)
+    vcpkg_add_to_path(${NASM_EXE_PATH})
+elseif (VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_find_acquire_program(GASPREPROCESSOR)
+    foreach(GAS_PATH ${GASPREPROCESSOR})
+        get_filename_component(GAS_ITEM_PATH ${GAS_PATH} DIRECTORY)
+        vcpkg_add_to_path(${GAS_ITEM_PATH})
+    endforeach(GAS_PATH)
 endif()
 
 set(LIBRARY_TYPE ${VCPKG_LIBRARY_LINKAGE})

--- a/ports/dav1d/vcpkg.json
+++ b/ports/dav1d/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "dav1d",
   "version": "1.4.0",
+  "port-version": 1,
   "description": "dav1d is a new open-source AV1 decoder developed by the VideoLAN and FFmpeg communities and sponsored by the Alliance for Open Media.",
   "homepage": "https://code.videolan.org/videolan/dav1d",
   "license": "BSD-2-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2178,7 +2178,7 @@
     },
     "dav1d": {
       "baseline": "1.4.0",
-      "port-version": 0
+      "port-version": 1
     },
     "daw-header-libraries": {
       "baseline": "2.106.0",

--- a/versions/d-/dav1d.json
+++ b/versions/d-/dav1d.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "c92b5d1be96a53ccc47eddb738e08111b537513f",
+      "version": "1.4.0",
+      "port-version": 1
+    },
+    {
       "git-tree": "0349d1196602b2983b1171647b30b5a9379920c6",
       "version": "1.4.0",
       "port-version": 0

